packages/lisk-p2p/src/p2p.ts

```diff
@@ -480,6 +497,52 @@ export class P2P extends EventEmitter {
 		return seedPeerUpdatedInfos;
 	}
 
+	private _pickRandomDiscoveredPeers(
+		count: number,
+	): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		const discoveredPeerList: ReadonlyArray<P2PDiscoveredPeerInfo> = [
+			...this._newPeers.values(),
+			...this._triedPeers.values(),
+		]; // Peers whose values has been updated atleast once.
+
+		return selectRandomPeerSample(discoveredPeerList, count);
+	}
+
+	private _handleGetAllPeersRequest(request: P2PRequest): void {
+		// TODO later: Remove fields that are specific to the current Lisk protocol.
+		const protocolPeerInfoList: ProtocolPeerInfoList = {
+			success: true,
+			peers: this._pickRandomDiscoveredPeers(MAX_PEER_LIST_BATCH_SIZE)
+				.map(
+					(peerInfo: P2PDiscoveredPeerInfo): ProtocolPeerInfo | undefined => {
+						const peerDiscoveredInfo = peerInfo;
+
+						if (!peerDiscoveredInfo) {
+							return undefined;
+						}
+						const { ipAddress, ...peerInfoWithoutIp } = peerDiscoveredInfo;
+
+						// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
+						return {
+							...peerInfoWithoutIp,
+							ip: ipAddress,
+							broadhash: peerInfoWithoutIp.broadhash
```

Comment:
do we need to overwrite these property?
it seems like if it exists, it will be deconstructed from `peerInfoWithoutIp`

Reply:
`ip` is by the name `ipAddress`, and `broadhash` and `nonce` are not required fields, they are unknown fields and we want to make sure to send these values as part of handshake.